targets/TARGET_Cypress/TARGET_PSOC6/PeripheralPins.c

```diff
@@ -0,0 +1,83 @@
+/*
+ * mbed Microcontroller Library
+ * Copyright (c) 2017-2018 Future Electronics
+ * Copyright (c) 2018-2019 Cypress Semiconductor Corporation
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stdlib.h>
+#include "PeripheralPins.h"
+
+static PinMap default_nc_pin_map = {NC, NC, 0};
+
+void try_create_pin_map(PinMap **pin_map, const cyhal_resource_pin_mapping_t* resource_map, size_t resource_size)
+{
+    if (*pin_map == NULL)
+    {
+        if (resource_size == sizeof(cyhal_resource_pin_mapping_t) && resource_map[0].pin == NC)
+        {
+            // If there are no valid connection, then use default NC pin map.
+            *pin_map = &default_nc_pin_map;
+        }
+        else {
+            *pin_map = (PinMap*)malloc(resource_size + sizeof(PinMap));
```

Comment:
We updated pin maps - static pin map. Why does this allocate on heap?

Reply:
There is already a static pin map located in mbed-os\targets\TARGET_Cypress\TARGET_PSOC6\psoc6csp\hal\include\pin_packages
This change to remove a second copy of the pin map.

I can't find a way to statically or globally create a copy of the map.
There is at most 1 copy of the pin maps because it check if the static pointer does not have an existing value before calling malloc.